### ISSUE TITLE
Support multiple servers

### ISSUE THREAD

**Issue by aminehmida** [gotify/android#28]
It will be awesome if the app can connect to multiple servers at once. Can be kinda of poor man high availability implementation:

A script tries to send a notification to `server1.com` if it's down it tries `server2.com` and because the android client is connected to both, you still get the notification.

Thank you.

**Comment by jmattheis** [comment]
Maybe it would be good, to have real support for high availability, with mirrored databases and stuff.

**Comment by sindastra** [comment]
I am in favour of having multi server and multi account support. Currently it's only possible to be signed in to one account@server.

**Comment by qdm12** [comment]
I came here to create this exact same issue 😄
I wish you could login on multiple servers at once and just merge all the notifications.

**Comment by reinethernal** [comment]
It would be really great to get pushes from multiple instances. For example, we have gotifyserver1 and gotifyserver2. They are all used for different needs, have different credentials, and so on. It would be nice to be able to receive messages from both independently (i.e. log in to gotifyserver1 and 2 at the same time and receive messages from both)

**Comment by fl0wfr** [comment]
Yes, it would be great to get notifications from several Gotify instances. That's the only feature missing that stops me from getting rid of Pushbullet (which is working as a service so I can push notifications wherever my hosted serviced are located).
In my case I need to be able to receive notifications from server2 when server1 doesn't have access to the Internet (server1 and server2 are not on the same location).

**Comment by damnwalledgarden** [comment]
I up vote too! If you have multiple servers and you monitor them with separate Gotify instances there is no way to check notifications on all of the instances.

P.S. For those who want a quick fix, you can recompile the Android app with a different name, e.g. com.github.gotify2 (etc) to run separate servers on each app.

**Comment by zethis** [comment]
Really needed in my case 👍

**Comment by pmontrasio** [comment]
Yes please, to be able to install gotify server for multiple customers and get notifications on my phone. I don't want to have to buy one phone per customer :-) so I'm not telling customers about gotify.

**Comment by MilorES** [comment]
I too would like. Is it the right site for developers to see?

**Comment by jmattheis** [comment]
@MilorES I don't really know what you mean with "order new features", but you certainly can request features on this issue tracker. It is pretty unlikely that I will implement this specific ticket, because I don't need it.

I'm open to accept contributions to this project, but as this feature is a little bigger, we would need to first discuss how this should be implemented.

**Comment by MilorES** [comment]
My apologies.

I did not know the difficulty of adding a connection to a second gotify server in the android application, I have seen applications like nextcloud for android that allow it and show you the server that you are connected to and/or send you the notification. (I understand that there must also be many more developers or people who contribute).

I would love to be able to contribute, but my level of English, since I use the translator (you should guess) or programming at the moment is not enough.

I only came up with several ways to do it (without knowing the code or how to program it).

The simplest one I saw (which doesn't have to be), that the android application has a connection to a second server and that it shows all the notifications it receives.
Let it be the APP, that is, the user who sends the notification that manages which gotify server to send it to.

**Comment by jmattheis** [comment]
No problem, but you also have to rework the login process to allow multiple users, but otherwise I agree with how it should work.

**Comment by tobinq** [comment]
i'ld appreciate this too.

it would be possible to connect to a private gotify server and one at work with the same phone while connecting those two by scripts or notification chains is rather not considerable.

however the approach to recompile with another name could help, so i thought maybe you could publish a preview version under a different name like "gotify-preview".

this could help in multiple ways. those who want two servers can just do so by installing both, while those who want to test new versions/features before updating the one used as "live" environment one could too.

anyway thanks for all !

**Comment by W1BTR** [comment]
Definitely would love this. I have a ntfy server for my uptimekuma at home. Now at my new job, we use ntfy, so I have to stop using my home server altogether!

**Comment by Park0** [comment]
I just saw this issue when scrolling to the list of issues. I wanted to make a remark about this because there is one thing in the discussion i am missing. 
For me battery usage is a thing. Having a failover server would be nice. But things as round robin DNS kan do a poor man failover. Why not chain gotify servers on server level. Send messages from server A to Server B to client. This would save complexity & battery usage on the client. (also on other implementations).

**Comment by eldiaboloz** [comment]
As I was researching how to receive notifications from multiple servers on my phone and had the same battery concerns for multiple active connections I decided to setup a [simple proxy script](https://gist.github.com/eldiaboloz/c743257855ed0b36a3ab07a8c0d81d05#file-gotify-proxy-sh) and connect to one instance (central) and proxy the notifications from all other instances to it.
If the central instance goes down, it stops working, so no HA with this setup.

**Comment by Nespouique** [comment]
Hi, I have exactly the same request, as I have 2 homelabs (one main and one backup) and the idea is that if one of them goes down, I can still receive alerts from the other one, and thus the need to connect android app to 2 differents gotify servers. The only workaround for now is to use another notification provider (discord for example)

**Comment by jmattheis** [comment]
You could use [Insular](https://f-droid.org/en/packages/com.oasisfeng.island.fdroid/) or [shelter](https://f-droid.org/packages/net.typeblog.shelter/), to clone the gotify app and login into a different gotify/server.

**Comment by shadowimmage** [comment]
I also would like there to be multiple user / server profiles in the app so that I can have a work server and my own homelab server both feed alerts to my phone. I'll check out shelter or insular as a potential workaround for now.
